src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/TestFiles/RazorSourceGeneratorComponentTests/ScriptTag_WithVariable/Views_Home_Index_8.html

```diff
@@ -0,0 +1,33 @@
+﻿<script>console.log('What&#x27;s up');</script>
+<div>console.log('What&#x27;s up');</div>
+<script>console.log('No variable');</script>
+<div>console.log('No variable');</div>
+<script>
+    console.log('What&#x27;s up');
+</script>
+<div>
+    console.log('What&#x27;s up');
+</div>
+<script>
+    console.log('No variable');
+</script>
+<div>
+    console.log('No variable');
+</div>
+Component:
+<script>console.log('What\u0027s up');</script>
+<div>console.log(&#x27;What&#x27;s up&#x27;);</div>
+<script>console.log('No variable');</script>
+<div>console.log('No variable');</div>
+<script>
+    console.log('What\u0027s up');
```

Comment:
Do the tests show somewhere what the Razor compiler output (in terms of C# source code) is? It would be helpful to know what `RenderTreeBuilder` APIs they are going to call for this input.

I'm asking because I don't really know how this `What\u0027s up` gets generated. Is that coming out of an HTML encoder or a JavaScript string encoder? How is the HTML itself being constructed - does the test call into Blazor code to get this, and if so, which of the Blazor renderers does it use?

For correctness, the `What\u0027s up` would need to come from a JS string encoder, not an HTML encoder.

Reply:
Added generated code baselines. For the Blazor component in .NET 8, see https://github.com/jjonescz/razor/blob/aspnetcore52547-ScriptBlockEncoding/src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/TestFiles/RazorSourceGeneratorComponentTests/ScriptTag_WithVariable/8/Shared_Component1_razor.g.cs

The `What\u0027s up` is generated via a call to `__builder.AddContent(msg);` where the Blazor static renderer encodes it via JavaScript encoder I think. The HTML is generated via a hacky (and maybe outdated) way to render CSHTML files:

https://github.com/dotnet/razor/blob/999b33452cdd3c4ac639e21419f4b34f2999571d/src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs#L111

The Blazor component is rendered from the CSHTML:

https://github.com/dotnet/razor/blob/5b1206a8d040bc3aa7510e2eb2c0108c9169ed8d/src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorComponentTests.cs#L617